src/main/java/touro/chess/RookPiece.java

```diff
@@ -0,0 +1,42 @@
+package touro.chess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RookPiece extends AbstractPiece
+{
+    private Object Move;
+
+    public RookPiece(Location location, PieceColor color)
+    {
+        super(location, color);
+    }
+
+    @Override
+    public List<Move> getMoves()
+    {
+        List<Move> List = new ArrayList<Move>();
+        Location curLocation = getLocation();
+        //for every row on the board adding one to the original row placement
+        for(int row=0; row < 9; row++)
+        {
+            Location newLoc = new Location(row, curLocation.getColumn());
+            //check if the row and col are not the same as original location if not add new move to list
```

Comment:
I check in isLegal() to make sure that a move does not move to its same square, so I don't think this is necessary

Reply:
It's fine to omit a Move to the same Location here.